service/history/execution/context.go

```diff
@@ -736,14 +757,23 @@ func (c *contextImpl) UpdateWorkflowExecutionWithNew(
 				newContext.Clear()
 			}
 		}()
-
 		newWorkflow, newWorkflowEventsSeq, err = newMutableState.CloseTransactionAsSnapshot(
 			now,
 			*newWorkflowTransactionPolicy,
 		)
 		if err != nil {
 			return err
 		}
+		if len(newWorkflow.WorkflowRequests) != 0 && len(currentWorkflow.WorkflowRequests) != 0 {
+			// TODO: convert it to an error if we've verified in production
```

Comment:
associate all TODO comments with a ticket id where folks can find more context and it will also help to avoid any relevant TODOs when we tackle that ticket in the future
```
// TODO(CDNC-123): ...
```